README.md

```diff
@@ -1,102 +1,262 @@
 
-Takes care of the complexities of the SAML protocol and provides an easy interface for using it. Specifically, creating metadata.xml files, creating `AuthnRequest`s and parsing and validating `AuthnResponse`s.
-
-This is exposed as both a series of functions that implement each step of the SAML protocol, and an Express middleware that creates the necessary endpoints for the metadata, the login and the assertion.
+`saml2-js` is a module that abstracts away the complexities of the SAML protocol and provides an easy interface for using it in node.
 
 ## Installation
 
 ```bash
-  npm install saml2-js
+  npm install saml2-js --save
 ```
 
-## Expected Usage
+## Usage
 
 Include the SAML library.
 
-```coffee
-  saml_lib = require('saml')
+```javascript
+  var saml2 = require('saml2-js');
 ```
 
-To use the saml library, we think in terms of service providers (e.g. Clever) and identity providers (e.g. partners that use ADFS).
+## Documentation
 
-```coffee
-  sp = saml_lib.service_provider
-    private_key : 'saml.pem'
-    certificate : 'saml.crt'
+This library exports two constructors.
 
-  idp = saml_lib.identity_provider
-    sso_login_url : 'https://www.example.com/login'
-    sso_logout_url : 'https://www.example.com/logout'
-    certificate : 'adfs.crt'
+- [`ServiceProvider`](#ServiceProvider) - Represents a service provider that relies on a trusted [`IdentityProvider`](#IdentityProvider) for authentication and authorization in the SAML flow.
+- [`IdentityProvider`](#IdentityProvider) - Represents an online service that authenticates users in the SAML flow.
 
-```
+<a name="ServiceProvider" />
+### ServiceProvider(options)
+Represents a service provider that relies on a trusted [`IdentityProvider`](#IdentityProvider) for authentication and authorization in the SAML flow.
+
+#### Options
+- `entity_id` - **Required** - Unique identifier for the service provider, often the url of the metadata file.
+- `private_key` - **Required** - Private key (PEM format string) for the service provider.
+- `certificate` - **Required** - Certificate (PEM format string) for the service provider.
+- `assert_endpoint` - **Required** - URL of service provider assert endpoint.
+- `force_authn` - Boolean that allows forcing re-authentication of users even if the user has a SSO session with the IdP.  This can also be configured on the [IdP](#IdentityProvider) or on a per-method basis.
+- `auth_context` - Specifies `AuthnContextClassRef`.  This can also be configured per-method basis.
+- `nameid_format` - Format for Name ID.  This can also be configured on a per-method basis.
+- `sign_get_request` -  Boolean, if true signs the request.  This can also be configured on the [IdP](#IdentityProvider) or on a per-method basis.
+- `allow_unencrypted_assertion` -  Boolean, when set to true allows unencrypted assertions.  This can also be configured on the [IdP](#IdentityProvider) or on a per-method basis.
```

Comment:
I want to be consistent with how we denote the types of options. In some options we don't say what type to provide (most are strings I think), in some we say upfront `Boolean - ...`, etc. 

One option: We can say before the options:

> Represents a service provider that relies on a trusted [`IdentityProvider`](#IdentityProvider) for authentication and authorization in the SAML flow. Takes the following options (strings unless specified otherwise):

and then just denote which options aren't strings.